pulp_ansible/app/modelresource.py

```diff
@@ -116,6 +116,19 @@ def before_import_row(self, row, **kwargs):
         col = Collection.objects.get(name=row["name"], namespace=row["namespace"])
         row["collection"] = str(col.pk)
 
+    def get_instance(self, instance_loader, row):
+        """
+        This fixes https://github.com/pulp/pulp_ansible/issues/1986
+        Find and set the current highest version to False before importing the new highest version.
+        """
+        instance = super().get_instance(instance_loader, row)
+        if not instance and row["is_highest"] == "1":
+            CollectionVersion.objects.filter(
+                collection_id=row["collection"], is_highest=True
+            ).update(is_highest=False)
+
+        return instance
```

Comment:
Oh damn it! `is_highest` really needs to go away.
Thank you for finding this.
I think however, we cannot be sure that when importing a highest collection version, that there is no higher one in the system already. Am I right, that you don't actually care about the `is_highest` flag? And this seems to fix the issue with the database constraint?
I guess we could also just lift the constraint, if we realize we want to accept inconsistent data here.